Automattic/wp-calypso
# Import: adopt the light import URL capture screen

#### Proposed Changes

* `client/blocks/import/capture` is renamed `client/blocks/import/capture-retired`
* Moved `client/blocks/import-light/capture` into `client/blocks/import/`
* Created a new component - `CaptureStep` for wrapping the following functions:
  * displaying the URL capture screen (Capture component)
  * state switching (init, scanning)
  * action - `analyzeUrl` integration
  * tracking events for state changes
* Switching between the retired URL capture screen and the new URL capture screen based on the feature flag - `onboarding/import-light-url-screen`
* Two new props created for the `Capture` component
  * hasError: make the parent component be able to decide if the `Capture` component should present the error validation state. We need this prop because the `Capture` component should display the validation error when the analyzeUrl API throws an error.
  * onInputChange: `resetError()` will be invoked when the user changes the input text, thus exposing this prop for handling the input change event

#### Testing Instructions

Testing with the feature flag `onboarding/import-light-url-screen` enabled

* Go to `http://calypso.localhost:3000/setup/import?siteSlug=<YOUR-SITE>&flags=onboarding/import-light,signup/goals-step-2`
* The new URL capture screen should appear

<img width="1118" alt="Screen Shot 2022-07-19 at 1 23 13 PM" src="https://user-images.githubusercontent.com/1024985/179671435-a8f09971-1700-47d3-8400-e8770eec55d6.png">

* Click **Don't have a site address?** link
* The import list screen should appear

<img width="1230" alt="Screen Shot 2022-07-14 at 11 18 44 PM" src="https://user-images.githubusercontent.com/1024985/179017578-286ecba6-0eed-457e-8d54-ce6ade37e20f.png">

* Click **Back** should go back to the URL capture screen
* Enter **wwww.goooo** in the **Existing site address** input box
* Click **Continue** button or hit **Enter**
* Form validation message should appear

<img width="1154" alt="Screen Shot 2022-07-14 at 11 25 48 PM" src="https://user-images.githubusercontent.com/1024985/179018937-ad09d537-77a0-4033-83b2-81800a3c07d8.png">

* Enter **oknalok.com** in the **Existing site address** input box
* Click **Continue** button or hit **Enter**
* The import-ready preview screen should appear

<img width="1774" alt="Screen Shot 2022-07-14 at 11 31 12 PM" src="https://user-images.githubusercontent.com/1024985/179020189-302cf499-5a97-4065-af4a-7521f9092c09.png">

* Click **Back** should go back to the URL capture screen
* Click **Back** again should go back to the goals capture screen

<img width="1056" alt="Screen Shot 2022-07-14 at 11 34 16 PM" src="https://user-images.githubusercontent.com/1024985/179020727-2826dca7-f88b-454d-a80b-f2782973d7da.png">

* Click **Continue**
* Enter **https://addyosmani.com/blog** in the **Existing site address** input box
* Click **Continue** button or hit **Enter**
* The import not ready screen should appear

<img width="691" alt="Screen Shot 2022-07-19 at 1 44 34 PM" src="https://user-images.githubusercontent.com/1024985/179674411-9778e442-1cc3-4edc-9a8b-ea5e5afd13f5.png">

* Click **Start building**
* Should display the goal capture screen with the import goal deselected

<img width="1143" alt="Screen Shot 2022-07-19 at 1 46 26 PM" src="https://user-images.githubusercontent.com/1024985/179674535-a88015fd-b271-4961-9294-d75f5c60693e.png">


Testing without the feature flag `onboarding/import-light-url-screen` enabled

* Go to `http://calypso.localhost:3000/setup/import?siteSlug=<YOUR-SITE>&flags=signup/goals-step-2`
* The original URL capture screen should appear

<img width="1404" alt="Screen Shot 2022-07-14 at 11 37 02 PM" src="https://user-images.githubusercontent.com/1024985/179021763-e0bc1d95-5c27-4870-a443-52dd69516ff8.png">

* Click **Don't have a site address?** link
* The import list screen should appear

<img width="1230" alt="Screen Shot 2022-07-14 at 11 18 44 PM" src="https://user-images.githubusercontent.com/1024985/179021917-8e72ac8b-321e-40a6-b543-add8a85f7221.png">

* Click **Back** should go back to the URL capture screen
* Enter **wwww.goooo** in the search input box
* Hit **Enter**
* Form validation message should appear

<img width="910" alt="Screen Shot 2022-07-14 at 11 40 40 PM" src="https://user-images.githubusercontent.com/1024985/179022079-4d763830-6b9d-4da8-92ba-c530b41f88c1.png">

* Enter **oknalok.com** in the **Existing site address** input box
* Click **Continue** button or hit **Enter**
* The import-ready preview screen should appear

<img width="1774" alt="Screen Shot 2022-07-14 at 11 31 12 PM" src="https://user-images.githubusercontent.com/1024985/179022213-1296d021-fbad-4fd0-9c52-fb269de126b8.png">

* Click **Back** should go back to the original URL capture screen
* Click **Back** again should go back to the goals capture screen

<img width="1056" alt="Screen Shot 2022-07-14 at 11 34 16 PM" src="https://user-images.githubusercontent.com/1024985/179022281-37edc50c-6868-4430-a219-822468e9eb1a.png">


#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.


Fixes #65336
